### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by jmcclory123** [brennanaba/ABlooper#7]
**Describe the bug**
I generated an scFv model in ABodyBuilder and remodelled the CDRs using ABlooper. 
I wanted to then add the side chains using openMM as described in the 'usage' section, however, upon running the following command:

ABlooper my_antibody_model.pdb --output ABlooper_model.pdb --side_chains  

I get a permission error related to 'pdbfixer.py;

C:\Users\James\Documents\Python_Training\ABlooper>ABlooper my_fv_model_rank1_imgt_scheme.pdb --output my_model_side_chains.pdb --side_chains
Warning: importing 'simtk.openmm' is deprecated.  Import 'openmm' instead.
Traceback (most recent call last):
  File "c:\users\james\anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\james\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\James\anaconda3\Scripts\ABlooper.exe\__main__.py", line 7, in <module>
  File "c:\users\james\anaconda3\lib\site-packages\ABlooper\command_line.py", line 40, in main
    predictor.write_predictions_in_pdb_format(output_file)
  File "c:\users\james\anaconda3\lib\site-packages\ABlooper\ABlooper.py", line 207, in write_predictions_in_pdb_format
    old_text = openmm_refine(old_text, self.CDR_with_anchor_slices)
  File "c:\users\james\anaconda3\lib\site-packages\ABlooper\openmm_refine.py", line 90, in openmm_refine
    output = relax_CDR_loops(pdb_txt, CDR_definitions, spring_constant)
  File "c:\users\james\anaconda3\lib\site-packages\ABlooper\openmm_refine.py", line 24, in relax_CDR_loops
    fixer = pdbfixer.PDBFixer(tmp.name)
  File "c:\users\james\anaconda3\lib\site-packages\pdbfixer\pdbfixer.py", line 243, in __init__
    file = open(filename, 'r')
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\James\\AppData\\Local\\Temp\\tmpnqynnp_p'

I have tried a few things to try and fix this. I updated, conda, anaconda and made sure that openMM was installed (tested and verified).

**To Reproduce**
1. Produce a model in ABodyBuilder using IMGT numbering
2. run the command: ABlooper my_antibody_model.pdb --output ABlooper_model.pdb --side_chains  

**Expected behavior**
Expect a model to be produced with the sidechains of the CDRs added and refined using openMM.

**Screenshots**
![ABlooper_Bug](https://user-images.githubusercontent.com/104768416/166247233-4fa29e9a-b2db-4de0-8cfb-e8bf7ab327ad.png)
![openMM TEST](https://user-images.githubusercontent.com/104768416/166247573-8eca211a-d47a-4b73-b41e-1b1766469c86.PNG)
![ABlooper_pdb_file](https://user-images.githubusercontent.com/104768416/166247827-8befc3a7-78c9-46bc-9a03-8d2e59f7b62e.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Python version 3.10.4
 - PyTorch version 1.11.0+cpu

**Additional context**

Any help with this would be really appreciated.

You can reach me directly by email: jmcclory123@gmail.com

**Comment by brennanaba** [comment]
Hi James,

Thanks a lot for reporting this. It appears to be a problem with creating tempfiles with python on Windows. 

I think I have fixed this now. Let me know if you have any other issues.

All the best,

Brennan

**Comment by jmcclory123** [comment]
Hi Brennan,

Many thanks for your quick response and solution.

I seem to be running into a new problem now in that the *--side_chains
*argument
is no longer recognized:

[image: image.png]

Your assistance is greatly appreciated.

Kind regards,
James

**Comment by brennanaba** [comment]
Hi James,

Your image didn't upload. 

As a quick check, are you sure that OpenMM and ABlooper are installed in the same environment?

All the best,

Brennan

**Comment by jmcclory123** [comment]
> Hi James,
> 
> Your image didn't upload.
> 
> As a quick check, are you sure that OpenMM and ABlooper are installed in the same environment?
> 
> All the best,
> 
> Brennan


Apologies, hopefully the images uploads this time around. I had a dig around to ensure that OpenMM and ABlooper are installed in the same environment and they appear to be.

![pip_list](https://user-images.githubusercontent.com/104768416/166311574-9cd907f0-91d4-4524-bb29-e6d7e2118404.PNG)

![side_chain-argument](https://user-images.githubusercontent.com/104768416/166311513-f5b611cb-eae5-40a8-bfcb-9b69207301b7.PNG)

It seems that the **--side_chains** argument isn't being recognized. It was being recognized previously.

Many thanks,

James

**Comment by brennanaba** [comment]
Hi James,

The code is set up so that if it can't run OpenMM or PyRosetta the --side-chain flag is unavailable. Therefore the problem has to do with how you installed OpenMM.

From the images you have shown it seems you have tried to install openmm and simtk using pip. This means you probably have installed these two packages:
- openmm: https://github.com/rmcgibbo/openmm-cmd
- simtk: https://github.com/scopetools/SIM_recon_python

Instead, to run ABlooper you have to install the official version of OpenMM from <a href="https://github.com/openmm/openmm">here</a>.

The easiest way to this is is using conda (more info <a href="http://docs.openmm.org/7.6.0/userguide/application/01_getting_started.html">here</a>)

```bash
$ conda install -c conda-forge openmm pdbfixer
```

All the best,

Brennan

**Comment by jmcclory123** [comment]
Hi Brennan,

That makes sense - I've been able to run the -side-chain tool successfully.

Thanks for all your help with this.

When I opened the structure with the side-chains resolved in Maestro, it appears that the bonds between the atoms are missing. I also tried opening the structure in Discovery Studio and can see that the side-chains are quite distorted.

![Maestro-Side-chains](https://user-images.githubusercontent.com/104768416/166438291-29e37ebc-0280-447e-ab81-4345aba63dfd.PNG)

![Discovery-Studio-Side-chains](https://user-images.githubusercontent.com/104768416/166438431-05919298-b5a5-48d7-86e5-8d2af54d847d.PNG)


I was wondering if you ever experienced anything like this when using OpenMM for the side-chain generation?

Again, apologies for any inconvenience caused.

Kind regards,
James
